sentry/src/main/java/io/sentry/util/MapObjectReader.java

```diff
@@ -0,0 +1,350 @@
+package io.sentry.util;
+
+import io.sentry.ILogger;
+import io.sentry.JsonDeserializer;
+import io.sentry.ObjectReader;
+import io.sentry.SentryLevel;
+import io.sentry.vendor.gson.stream.JsonToken;
+import java.io.IOException;
+import java.util.AbstractMap;
+import java.util.ArrayDeque;
+import java.util.Date;
+import java.util.Deque;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@SuppressWarnings("unchecked")
+public final class MapObjectReader implements ObjectReader {
```

Comment:
You could add a simple test where you go full circle: e.g. serialize a context to a map, and the deserialize it back and see if it matches

Reply:
yeah I was wrapping it up yday https://github.com/getsentry/sentry-java/pull/3215/commits/1e76fc7499e8a1f4747c67c3aed4d1547c83759e